client/src/Pages/ProductOverviewPage.jsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import CurrentPageNav from "../components/CurrentPageNav";
+import ImageSelection from "../components/ImageSelection";
+import ProductDetails from "../components/ProductDetails";
+import { getProductById } from "../utilities/productsApi";
+import NotFoundPage from "./NotFoundPage";
+
+const ProductOverviewPage = () => {
+  const [product, setProduct] = useState("");
+  const [images, setImages] = useState([]);
+  const [image, setImage] = useState("");
+  const [timeLeft, setTimeLeft] = useState("");
+  const { id } = useParams();
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await getProductById(id);
+        setProduct(response.data);
+        setImages(response.data.images.split(","));
+        setImage(response.data.images.split(",")[0]);
+        calculateTimeLeft(response);
+      } catch (error) {
+        console.error(error);
+      }
+    })();
+  }, []);
+
+  const calculateTimeLeft = (response) => {
+    const startDate = new Date();
+    const endDate = new Date(response.data.endDate);
+    const diffDays = Math.ceil(
+      Math.abs(endDate - startDate) / (1000 * 60 * 60 * 24)
+    );
+    const diffWeeks = Math.floor(diffDays / 7);
+    const diff = {
+      weeks: diffWeeks,
+      days: Math.floor(diffDays % 7),
+    };
+    setTimeLeft(diff);
+  };
+
+  return product ? (
+    <>
+      <CurrentPageNav title={product.productName} />
+      <div className="mx-40 mt-8 2xl:mx-72 flex">
+        <ImageSelection images={images} image={image} setImage={setImage} />
```

Comment:
Since image and setImage are only provided to ImageSelection component, and not used anywhere else in this component, we can delete them from ProductOverviewPage. 
`<ImageSelection images={images} />` should be enough, and inside ImageSelection you can handle currently displayed image

Reply:
setImage is being used in useEffect hook on that page